compiler/rustc_hir_analysis/src/lib.rs

```diff
@@ -214,18 +214,23 @@ pub fn check_crate(tcx: TyCtxt<'_>) -> Result<(), ErrorGuaranteed> {
     Ok(())
 }
 
-/// A quasi-deprecated helper used in rustdoc and clippy to get
-/// the type from a HIR node.
-pub fn hir_ty_to_ty<'tcx>(tcx: TyCtxt<'tcx>, hir_ty: &hir::Ty<'tcx>) -> Ty<'tcx> {
+/// Helper to lower a [`hir::Ty`] to a [`Ty`].
```

Comment:
Add back the mention of rustdoc and Clippy. I originally removed it since there were some uses in `rustc_privacy` until very recently. Now the note makes sense again.